File: Week3Selenium/src/test/java/pages/LoginPage.java
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class LoginPage extends BasePage {

    private static final By USERNAME = By.id("user-name");
    private static final By PASSWORD = By.id("password");
    private static final By LOGIN_BUTTON = By.id("login-button");
    private static final By ERROR_MESSAGE = By.cssSelector("[data-test='error']");

    public LoginPage(WebDriver driver) {
        super(driver);
    }

    public LoginPage open() {
        driver.get("http://localhost:5173/login");
        return this;
    }

    public CatalogPage login(String username, String password) {
        type(USERNAME, username);
        type(PASSWORD, password);
        click(LOGIN_BUTTON);
        return new CatalogPage(driver);
    }

    public String errorMessage() {
        return text(ERROR_MESSAGE);
    }
}

File: Playwright_Final_Capstone/pages/loginPage.ts
import { Page } from '@playwright/test';
import { BasePage } from './basePage';

export class LoginPage extends BasePage {
  constructor(page: Page) {
    super(page);
  }

  private readonly loginLink = () => this.page.getByRole('link', { name: /log in/i }).first();
  private readonly emailBox = () => this.page.getByLabel(/email/i).first();
  private readonly passwordBox = () => this.page.getByLabel(/password/i).first();
  private readonly signInButton = () => this.page.getByRole('button', { name: /sign in|log in/i }).first();
  private readonly errorBanner = () => this.page.getByRole('alert').filter({ hasText: /invalid|error/i }).first();

  async open(): Promise<void> {
    await this.navigate();
  }

  async login(email: string, password: string): Promise<void> {
    await this.loginLink().click();
    await this.emailBox().fill(email);
    await this.passwordBox().fill(password);
    await this.signInButton().click();
  }

  errorMessage() {
    return this.errorBanner();
  }
}
